docs/glossary.md

```diff
@@ -8,11 +8,17 @@ has_children: false
 
 # Glossary
 
+{% include toc.html %}
+
 ### Auditing
 Data auditing is data assessment to ensure its accuracy, security, and efficacy for specific usage. It also involves assessing data quality through its lifecycle and understanding the impact of poor quality data on the organization's performance and revenue. Ensuring data reproducibility, auditability, and governance is one of the key concerns of data engineers today.
 
 ### Branch
-A branch is a mutable pointer to a commit and its staging area (i.e., mutable storage where developers can create, update, and delete objects). When a user creates a commit from a branch, all the files from the staging area will be merged into the contents of the current branch, generating a new set of objects. The pointer is updated to reference the new set of objects. The new branch tip is set to the latest commit and the previous branch tip serves as the commit's parent. Learn more about the lakeFS branching model [here](https://docs.lakefs.io/understand/branching-model.html).
+
+Branches are similar in concept to Git branches.
+When creating a new branch, we are actually creating a consistent snapshot of the entire repository, which is isolated from other branches and their changes.
+
+A branch is a mutable pointer to a commit and its staging area (i.e., mutable storage where developers can create, update, and delete objects). When a user creates a commit from a branch, all the files from the staging area will be merged into the contents of the current branch, generating a new set of objects. The pointer is updated to reference the new set of objects. The new branch tip is set to the latest commit and the previous branch tip serves as the commit's parent. Learn more about the lakeFS branching model [here]({{ site.baseurl }}/understand/branching-model.html).
```

Comment:
"Developers" here confuses me.  **Anything** can create objects on a branch: programs, notebooks, manual uploads... whatever.  I am not even sure it is typically _developers_ who (directly) create objects.

Reply:
We do not intend to replace the object model/branching model page. Glossary can be an index of all terms related to lakeFS in alphabetical order to make it easily readable/searchable. We will have internal links from glossary to object model & branching model pages as needed.

Reply:
Later, glossary page would be extended to include all terms related to data engineering as well.